prow/kube/types.go

```diff
@@ -75,15 +75,18 @@ type PodStatus struct {
 }
 
 type Volume struct {
-	Name        string             `json:"name,omitempty"`
-	Secret      *SecretSource      `json:"secret,omitempty"`
-	DownwardAPI *DownwardAPISource `json:"downwardAPI,omitempty"`
-	HostPath    *HostPathSource    `json:"hostPath,omitempty"`
-	ConfigMap   *ConfigMapSource   `json:"configMap,omitempty"`
+	Name string `json:"name,omitempty"`
+	VolumeSource
```

Comment:
```go
VolumeSource `json:",inline"`
```

Reply:
actually this doesn't exist right? reverting
https://github.com/golang/go/issues/6213#issuecomment-308484515

Reply:
actually will investigate more in a bit.
/hold x2

Reply:
We are using https://github.com/ghodss/yaml which supports it.

https://godoc.org/gopkg.in/yaml.v2#Marshal

Reply:
hmm ok, I'll leave it then.